pipelines/src/main/scala/dagr/pipelines/ContaminateAndEvaluateVBID.scala

```diff
@@ -0,0 +1,135 @@
+ @arg(flag = "r", doc = "Reference fasta.") val ref: PathToFasta,
+ @arg(flag = "o", doc = "Output directory.") val out: DirPath,
+ @arg(flag = "t", doc = "Target regions.") val targets: PathToIntervals,
+ @arg(flag = "p", doc = "Output file prefix.") val prefix: String,
+ @arg(flag = "c", doc = "Contamination levels to evaluate.") val contaminations: Seq[Double],
+ @arg(flag = "d", doc = "Depth values to evaluate.") val depths: Seq[Integer],
+ @arg(flag = "m", doc = "Number of markers to use.") val markers: Seq[Integer]
+
+) extends Pipeline(Some(out)) {
+
+  override def build(): Unit = {
+    val strat = Some(DownsamplingStrategy.HighAccuracy)
+
+    val bam = out.resolve(prefix + ".bam")
+    val tmpBam = out.resolve(prefix + ".tmp.bam")
+    val metricsPrefix: Some[DirPath] = Some(out.resolve(prefix))
+    Files.createDirectories(out)
+    val bamYield = new mutable.HashMap[PathToBam, Int]
+
+    bams.map(bam => {
+      val metricPath: Path = out.resolve(prefix + bam.getFileName + ".qualityYieldMetrics")
+      val hsMetrics = new CollectHsMetrics(in = bam, ref = ref, targets = targets, prefix = Some(out.resolve(prefix)))
+
+      val fetchMedian = new FetchMedianCoverage(metricPath)
+      root ==> hsMetrics ==> fetchMedian ==> SimpleInJvmTask(bamYield.put(bam, fetchMedian.medianCoverage.getOrElse(0)))
+    })
+
+    val pairsOfBam = for (x <- bams; y <- bams) yield (x, y)
+
+    pairsOfBam.foreach { case (x, y) => {
+      contaminations.foreach(c => {
+        val dsX = out.resolve(prefix + x.getFileName + ".forCont." + c + ".tmp.bam")
+        val dsY = out.resolve(prefix + y.getFileName + ".forCont." + c + ".tmp.bam")
+        // create a mixture of c x + (1-c) y taking into account the depths of
+        // x and y
+        // D_x and D_y are the effective depths of x and y
+        // if we are to downsample x we need x to be downsampled to c/(1-c)
+        // if we are to downsample y we need y to be downsampled to (1-c)/c
+        // of of those two values will be less than 1.
+        val downsampleX = new DownsampleSam(in = x, out = dsX, proportion = c / (1 - c), strategy = strat).requires(Cores(1), Memory("32g"))
+        val downsampleY = new DownsampleSam(in = x, out = dsY, proportion = (1 - c) / c, strategy = strat).requires(Cores(1), Memory("32g"))
+
+        val copyX = new LinkFile(x, dsX)
+        val copyY = new LinkFile(y, dsY)
+        val downsampleX_Q = c / (1 - c) <= 1
+        val downsample = EitherTask.of(
+          downsampleX :: copyY :: Nil,
+          downsampleY :: copyX :: Nil,
+          downsampleX_Q)
+
+        val xFactor = if (downsampleX_Q) c / (1 - c) else 1
+        val yFactor = if (downsampleX_Q) 1 else (1 - c) / c
+
+        val resultantDepth: Double = bamYield.getOrElse(x, 0) * xFactor + bamYield.getOrElse(y, 0) * yFactor
+
+        val xyContaminatedBam = out.resolve(prefix + "__" + x.getFileName + "__" + y.getFileName +
+          "__" + c + ".bam")
```

Comment:
why double underscores?  also, use string interpolation.
```suggestion
        val xyContaminatedBam = out.resolve(f"${prefix}__${x.getFileName}__${y.getFileName}__${c}.bam")
```